### PR TITLE
Rendering labels (POI name etc.) directly to scene buffer

### PR DESCRIPTION
**Pull request by peterAdams1014**
Resolves [AB#1199](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/1199)

### CODE REVIEW

File: src/rendering/vcLabelRenderer.cpp [hunk 1]
```diff
@@ -1,40 +1,142 @@
 #include "vcLabelRenderer.h"
 #include "udPlatformUtil.h"
+#include "gl/vcMesh.h"
+#include "gl/vcRenderShaders.h"
+#include "vcInternalModels.h"
 
 #include "imgui.h"
+#include <gl\vcGLState.h>
```
Comment (samsinsane) [inline src/rendering/vcLabelRenderer.cpp]: Forward slash for compatibility with every non-Windows target we have.

File: src/rendering/vcLabelRenderer.cpp [hunk 2]
```diff
@@ -1,40 +1,142 @@
...
+  } everyObject;
+} gShader;
+
+vcMesh *g_LabelMesh = nullptr;
+
+static int gRefCount = 0;
+udResult vcLabelRenderer_Init()
+{
+  udResult result;
+  gRefCount++;
```
Comment (samsinsane) [inline src/rendering/vcLabelRenderer.cpp]: This uses postfix, the decrement uses prefix. Additionally, is this intended to be thread safe?

File: src/vcRender.cpp [hunk 3]
```diff
@@ -1203,8 +1218,9 @@ void vcRender_RenderScene(vcState *pProgramState, vcRenderContext *pRenderContex
 
   vcRenderSkybox(pProgramState, pRenderContext); // Drawing skybox after opaque geometry saves a bit on fill rate.
   vcRenderTerrain(pProgramState, pRenderContext);
-
+
```
Comment (samsinsane) [inline src/vcRender.cpp]: Revert this.

File: src/rendering/vcLabelRenderer.cpp (hunk 2 above)
Comment (peterAdams1014) [reply src/rendering/vcLabelRenderer.cpp]: No this is not mean to be thread safe. I have fixed the prefix issue.

**Pull request merged**